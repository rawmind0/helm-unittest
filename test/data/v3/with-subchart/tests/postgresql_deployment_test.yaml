suite: test postgresql deployment
templates:
  - charts/postgresql/templates/deployment.yaml 
tests:
  - it: should pass all kinds of assertion for subchart
    values:
      - ./values/image.yaml
    asserts:
      - isKind:
          of: Deployment
      - isAPIVersion:
          of: extensions/v1beta1
      - hasDocuments:
          count: 1
      - matchSnapshot:
          path: spec
