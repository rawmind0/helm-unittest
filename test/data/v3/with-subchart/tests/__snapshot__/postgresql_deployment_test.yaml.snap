should pass all kinds of assertion for subchart:
  1: |
    template:
      metadata:
        labels:
          app: RELEASE-NAME-postgresql
      spec:
        containers:
        - env:
          - name: POSTGRES_USER
            value: postgres
          - name: PGUSER
            value: postgres
          - name: POSTGRES_DB
            value: ""
          - name: POSTGRES_INITDB_ARGS
            value: ""
          - name: PGDATA
            value: /var/lib/postgresql/data/pgdata
          - name: POSTGRES_PASSWORD
            valueFrom:
              secretKeyRef:
                key: postgres-password
                name: RELEASE-NAME-postgresql
          - name: POD_IP
            valueFrom:
              fieldRef:
                fieldPath: status.podIP
          image: postgres:9.6.2
          imagePullPolicy: ""
          livenessProbe:
            exec:
              command:
              - sh
              - -c
              - exec pg_isready --host $POD_IP
            failureThreshold: 6
            initialDelaySeconds: 60
            timeoutSeconds: 5
          name: RELEASE-NAME-postgresql
          ports:
          - containerPort: 5432
            name: postgresql
          readinessProbe:
            exec:
              command:
              - sh
              - -c
              - exec pg_isready --host $POD_IP
            initialDelaySeconds: 5
            periodSeconds: 5
            timeoutSeconds: 3
          resources:
            requests:
              cpu: 100m
              memory: 256Mi
          volumeMounts:
          - mountPath: /var/lib/postgresql/data/pgdata
            name: data
            subPath: postgresql-db
        volumes:
        - name: data
          persistentVolumeClaim:
            claimName: RELEASE-NAME-postgresql
